src/golibmc.go

```diff
@@ -921,14 +1325,73 @@ func (client *Client) Stats() (map[string](map[string]string), error) {
 
 // Quit will close the sockets to each memcached server
 func (client *Client) Quit() error {
-	client.lock()
-	defer client.unlock()
+	client.lk.Lock()
+	if client.closed {
+		client.lk.Unlock()
+		return nil
+	}
+
+	close(client.openerCh)
+	if client.cleanerCh != nil {
+		close(client.cleanerCh)
+	}
+	for _, cr := range client.connRequests {
+		close(cr)
+	}
+	client.closed = true
+	client.lk.Unlock()
+	for _, cn := range client.freeConns {
+		if err := cn.quit(); err != nil {
```

Comment:
suppose caller would always handle the err and call Quit again?
Or just go on to free them all ignoring the err (hold and return at last )?

Reply:
Thanks. I think that ignoring the quit error is better. Fixed.
https://github.com/douban/libmc/pull/57/commits/aba5a7b279cfe026e484b14ff8efd90e829e171d